pkg/prometheus/statefulset.go

```diff
@@ -680,27 +680,32 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 			thanosImage = *p.Spec.Thanos.Image
 		}
 
+		thanosVersion := semver.MustParse(strings.TrimPrefix(*p.Spec.Thanos.Version, "v"))
+		clusterSupport := thanosVersion.LT(semver.MustParse("0.5.0-rc.0"))
```

Comment:
I personally think we should just ignore this. We'll just support v0.5.0 and above from here. For exactly these reasons we marked this experimental.

Reply:
There have been a lot of changes to thanos, but prometheus-operator still supports v0.1.0, but I agree that we should limit backwards compatibility. My personal preference might be to support the latest and previous minor versions.

How about we move forward with this less controversial change, and I'll prepare a follow-up that changes the ThanosSpec (deprecated options) and removes all backwards compatibility? Then that change can be debated and move slower.

Reply:
sounds good to me